kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1006,28 +1009,42 @@ private void writeSnapshot(long sid, XmlComponentConfigurations conf) throws Kur
             throw new KuraException(KuraErrorCode.CONFIGURATION_SNAPSHOT_NOT_FOUND);
         }
 
-        // Marshall the configuration into an XML
-        String xmlResult;
-        xmlResult = marshal(conf);
-        if (xmlResult == null || xmlResult.trim().isEmpty()) {
-            throw new KuraException(KuraErrorCode.INVALID_PARAMETER, conf);
+        if (fSnapshot.isDirectory()) {
+            throw new KuraException(KuraErrorCode.CONFIGURATION_ERROR, fSnapshot.getName() + "is not a file.");
+        }
+
+        File tempSnapshotFile;
+        try {
+            tempSnapshotFile = File.createTempFile(fSnapshot.getName(), "", new File(fSnapshot.getParent()));
```

Comment:
Please create a temporary file in the same directory as the destination snapshot, to enable moving it to the final location with an atomic move operation, using an appropriate naming scheme so that it will be ignored by reading code